File: 2015/go/day03/part1.go
package main

import "strconv"

func processPart1(input string) int {
	x, y := 0, 0
	visited := make(map[string]bool)
	visited["0,0"] = true

	for _, c := range input {
		switch c {
		case '^':
			y++
		case 'v':
			y--
		case '>':
			x++
		case '<':
			x--
		}

		visited[strconv.Itoa(x)+","+strconv.Itoa(y)] = true
	}
	return len(visited)
}

File: 2015/deno/day03/part-1.ts
export function processPart1(input: string): number {
	const trimmed = input.trim();
	const directions = trimmed.split('');

	const visited = new Set<string>();
	let x = 0;
	let y = 0;
	visited.add(`${x},${y}`);

	for (const direction of directions) {
		switch (direction) {
			case '^':
				y++;
				break;
			case 'v':
				y--;
				break;
			case '>':
				x++;
				break;
			case '<':
				x--;
				break;
		}
		visited.add(`${x},${y}`);
	}
	return visited.size;
}
